### PR TITLE
[bpo-37433] Fix `SyntaxError` indicator printing too many spaces for multi-line strings

### PR DESCRIPTION
**Pull request by asottile**
https://bugs.python.org/issue37433

### CODE REVIEW

File: Lib/test/test_cmd_line_script.py [hunk 1]
```diff
@@ -613,6 +613,20 @@ def test_syntaxerror_indented_caret_position(self):
             self.assertNotIn("\f", text)
             self.assertIn("\n    1 + 1 = 2\n    ^", text)
 
+    def test_syntaxerror_multi_line_fstring(self):
```
Comment (terryjreedy) [inline Lib/test/test_cmd_line_script.py:616]: Perhaps
```suggestion
    def test_syntaxerror_multi_line_fstring(self):  # bpo-37433
```

**Review by terryjreedy — COMMENTED** [review]
Using the example in the new test, I verified the problem and fix on Win 10.

I am not familiar enough with C and CPython C code to be sure that fix is *exactly* what is needed.

File: Lib/test/test_cmd_line_script.py (hunk 1 above)
Comment (asottile) [reply Lib/test/test_cmd_line_script.py:616]: in previous patches I was told to remove the `bpo` comment as it's essentially noise (and `git blame` can retrieve the same information) -- let me know if you'd like to follow that suggestion or this one 👍

**Comment by asottile** [comment]
> 
> 
> Using the example in the new test, I verified the problem and fix on Win 10.
> 
> I am not familiar enough with C and CPython C code to be sure that fix is _exactly_ what is needed.

to explain a little better in words what's happening in this block:

- initially there is a buffer which is allocated, `multi_line_start` and `line_start` point at the beginning of that lines
- at some point, the tokenizer realises it needs to allocate more and read more to fill out the buffer for a multi line token
- this causes a reallocation of the original buffer into a new buffer, all of the pointers are copied over and adjusted for the new positions and offset -- however the `mutli_line_start` was not being copied and adjusted -- this patch fixes that bit

File: Lib/test/test_cmd_line_script.py (hunk 1 above)
Comment (terryjreedy) [reply Lib/test/test_cmd_line_script.py:616]: Leave it alone for now.

**Review by vstinner — APPROVED** [review]
LGTM. But I would prefer that at least another core dev reviews the change.

**Review by pablogsal — APPROVED** [review]
LGTM

I checked the examples provided and `tok->multi_line_start` does not point anymore to random memory (the one that is there before the `PyMem_REALLOC` happens).

I would suggest adding more tests for this that do not involve f-strings, maybe in a different PR.

**Comment by pablogsal** [comment]
@asottile  Does this affect 3.7?

**Comment by asottile** [comment]
@pablogsal nope, only 3.8+

**Comment by pablogsal** [comment]
Thank you very much @asottile for the fix :)

**Pull request merged**